### PR TITLE
M3-5458: Adjustments for API Scheduled Maintenance feature flag and banner

### PR DESCRIPTION
**Pull request by DevDW**
## Description
The feature flag for API Maintenance events needed to be adjusted to give us more control over the title and body of the banner due to length and specificity concerns of the Statuspage data within the context of Cloud.

This PR accounts for that adjustment and does some other minor cleanup to the banner.

Screen Shot 2021-09-09 at 11 30 37 AM

## How to test
Pointing to prod, you should see a global banner that corresponds to [this upcoming maintenance](https://status.linode.com/incidents/khfqpfsfdh0g). Persisted dismissibility should still work across pages within the app and across different sessions.

You can uncomment the mocked data and turn on the MSW to confirm multiple maintenances are handled again.

### CODE REVIEW

**Review by ndukatz — COMMENTED** [review]
Looks good! showing throughout the site and dismissing persists through pages and sessions.

**Review by ndukatz — APPROVED** [review]

File: packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx [hunk 1]
```diff
@@ -43,24 +54,40 @@ export const APIMaintenanceBanner: React.FC<Props> = (props) => {
...
+    const correspondingSuppliedMaintenance = suppliedMaintenances.find(
+      (apiMaintenanceEvent) =>
+        apiMaintenanceEvent.id === scheduledAPIMaintenance.id
+    );
+
+    const bannerTitle =
+      correspondingSuppliedMaintenance?.title !== undefined &&
+      correspondingSuppliedMaintenance?.title !== ''
+        ? correspondingSuppliedMaintenance.title
+        : scheduledAPIMaintenance.name;
```
Comment (HanaXu) [inline packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx:71]: What about
```typescript
const bannerTitle = correspondingSuppliedMaintenance?.title || scheduledAPIMaintenance.name;
```

File: packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx [hunk 2]
```diff
@@ -43,24 +54,40 @@ export const APIMaintenanceBanner: React.FC<Props> = (props) => {
...
+
+    const bannerBody =
+      correspondingSuppliedMaintenance?.body !== undefined &&
+      correspondingSuppliedMaintenance?.body !== ''
+        ? correspondingSuppliedMaintenance?.body
+        : mostRecentUpdate.body;
```
Comment (HanaXu) [inline packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx:77]: Same as above

File: packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx [hunk 3]
```diff
@@ -43,24 +54,40 @@ export const APIMaintenanceBanner: React.FC<Props> = (props) => {
...
+
     return (
       <Notice important warning dismissible onClose={onDismiss}>
```
Comment (HanaXu) [inline packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx:80]: We should add a unique key to the Notice:

image

**Review by HanaXu — APPROVED** [review]
LGTM 🔨 , just some minor comments!

File: packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx (hunk 3 above)
Comment (DevDW) [reply packages/manager/src/features/GlobalNotifications/APIMaintenanceBanner.tsx:80]: Good catch, thanks!

**Pull request merged**